### ISSUE TITLE
save -u  commits moved annexed files without adjusting their symlinks

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1418]
whenever annexed files are moved around, git annex adjusts paths in .git/hooks/pre-commit.
and that is happening just fine after the move if we just do 'git commit' or 'datalad save'

But if I do `datalad save -u` those with incorrect symlinks are committed and nothing gets fixed upon commit (I do  not see 'git commit' running, so I guess that is done via GitPython's commit?) and we ending up with broken symlinks all around... heh

correct of plain `save` (doesn't do any 'git annex add')
```shell
$> builtin cd /tmp/; chmod +w -R /tmp/testl; rm -rf /tmp/testl; datalad create /tmp/testl && cd /tmp/testl && mkdir d && echo 123 > d/123 && datalad add d/123 && git mv d/123 123 && datalad save && ls -l 123      
[INFO   ] Creating a new annex repo at /tmp/testl 
Created dataset at /tmp/testl.                                                                                                                                                                            
W: git-annex repositories not (yet) supported in the prompt
Added /tmp/testl/d/123                                                                                                                                                                                    
Saved state: d565e942c8f9cd7aa4e9ed8b3daf22f1a206ac61 for <Dataset path=/tmp/testl>
lrwxrwxrwx 1 yoh yoh 108 Mar 22 14:29 123 -> .git/annex/objects/pF/Zf/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f
```

incorrect operation of save -u:
```shell
$> chmod +w -R /tmp/testl; rm -rf /tmp/testl; datalad create /tmp/testl && cd /tmp/testl && mkdir d && echo 123 > d/123 && datalad add d/123 && git mv d/123 123 && datalad save -u && ls -l 123 
[INFO   ] Creating a new annex repo at /tmp/testl 
Created dataset at /tmp/testl.                                                                                                                                                                            
W: git-annex repositories not (yet) supported in the prompt
Added /tmp/testl/d/123                                                                                                                                                                                    
Saved state: b5c7e86e4d3e6f67479469152020e0c5dc5f8bb1 for <Dataset path=/tmp/testl>
lrwxrwxrwx 1 yoh yoh 111 Mar 22 14:28 123 -> ../.git/annex/objects/pF/Zf/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f
```

.... it is all due to `git commit /tmp/testl` -- somehow that causes git-annex to not fixup the links
```shell
$> builtin cd /tmp/; chmod +w -R /tmp/testl; rm -rf /tmp/testl; mkdir /tmp/testl  && cd /tmp/testl && git init && git annex init && mkdir d && echo 123 > d/123 && datalad add d/123 && git mv d/123 123 && git annex add . && git commit -m msg /tmp/testl && ls -l 123 && git status       
Initialized empty Git repository in /tmp/testl/.git/
init  ok
(recording state in git...)
('committing ', '[DATALAD] added content', ['/tmp/testl/d/123'])                                                                                                                                          
Added /tmp/testl/d/123
[master 7b86ef7] msg
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename d/123 => 123 (100%)
lrwxrwxrwx 1 yoh yoh 181 Mar 22 14:45 123 -> ../.git/annex/objects/G6/qW/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b/SHA256E-s4--181210f8f9c779c26da1d9b2075bde0127302ee0e3fca38c9a83f5b1dd8e5d3b
On branch master
nothing to commit, working tree clean
```
filed against annex: http://git-annex.branchable.com/bugs/annex_doesn__39__t_fixup_symlinks_when___34__git_commit_path__95__to__95__repo__34___is_used/?updated

but I think we should be able to address it by simply running commit in that directory instead of providing directory's path in the cmdline

PS use of 'swallow_logs' within code (not in tests) makes it really harder to debug when something goes wrong since then debug messages are swallowed! :-/

**Comment by mih** [comment]
FTR: current master (fa39d2fb9d5e75647ac2bf78f40b7e90ecc2c991) shows the expected behavior in both cases. with and without `-u`.

```
(datalad3-dev) 130 mih@meiner /tmp % builtin cd /tmp/; chmod +w -R /tmp/testl; rm -rf /tmp/testl; datalad create /tmp/testl && cd /tmp/testl && mkdir d && echo 123 > d/123 && datalad save d/123 && git mv d/123 123 && datalad save && ls -l 123 
chmod: cannot access '/tmp/testl': No such file or directory
[INFO   ] Creating a new annex repo at /tmp/testl 
create(ok): /tmp/testl (dataset)                                                             
add(ok): d/123 (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  save (ok: 1)
save(ok): . (dataset)
lrwxrwxrwx 1 mih mih 108 Dec 10 12:51 123 -> .git/annex/objects/pF/Zf/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f

(datalad3-dev) mih@meiner /tmp/testl (git)-[master] % builtin cd /tmp/; chmod +w -R /tmp/testl; rm -rf /tmp/testl; datalad create /tmp/testl && cd /tmp/testl && mkdir d && echo 123 > d/123 && datalad save d/123 && git mv d/123 123 && datalad save -u && ls -l 123
[INFO   ] Creating a new annex repo at /tmp/testl 
create(ok): /tmp/testl (dataset)                                                             
add(ok): d/123 (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  save (ok: 1)
save(ok): . (dataset)
lrwxrwxrwx 1 mih mih 108 Dec 10 12:52 123 -> .git/annex/objects/pF/Zf/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f/MD5E-s4--ba1f2511fc30423bdbb183fe33f3dd0f

```

**Comment by mih** [comment]
The release of 0.12 with this fix is imminent (#3993) closing.
